src/foam/java/refinements.js

```diff
@@ -233,7 +233,7 @@ if ( ! ${foam.java.asJavaValue(vp.predicate)}.f(obj) ) {
     {
       name: 'asJavaValue',
       code: function() {
-        return `${this.sourceCls_.id}.${foam.String.constantize(this.name)}`;
+        return `${this.forClass_}.${foam.String.constantize(this.name)}`;
```

Comment:
This is the reason why I started down this path. sourceCls_ wasn't coming through as I was expecting it to but I shouldn't have been using that property in the first place since it's technically not a property and shouldn't be relied upon.